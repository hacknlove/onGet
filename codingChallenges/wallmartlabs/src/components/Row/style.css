.row {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.row-title > div {
  line-height: 14px;
}

.grand-total {
  margin-top: 10px;
  margin-bottom: 10px;
  font-size: 1.25em;
  padding-top: 10px;
  padding-bottom: 10px;
  font-weight: bold;
  border-top: solid 1px  #e6e7e8;
  border-bottom: solid 1px  #e6e7e8;
}

.row-figure {
  font-weight: bold;
}
