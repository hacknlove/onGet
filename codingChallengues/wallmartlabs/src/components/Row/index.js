import React, { PureComponent } from 'react';
import './style.css';
import PropTypes from 'prop-types';

export default class Row extends PureComponent {
  render() {
    return (
      <div className={this.props.total ? 'row grand-total' : 'row'}>
        {this.props.toolTip ? (
          this.props.toolTip
        ) : (
          <div className="row-title">
            {this.props.title}
            {this.props.children}
          </div>
        )}
        <div
          style={
            this.props.less === true ? { color: 'red' } : { color: 'black' }
          }
          className="row-figure"
        >
          {this.props.less ? '-' : ''}${this.props.figure}
        </div>
      </div>
    );
  }
}

Row.propTypes = {
  total: PropTypes.bool,
  less: PropTypes.bool,
  toolTip: PropTypes.element,
  children: PropTypes.element,
  figure: PropTypes.number,
  title: PropTypes.string
};
