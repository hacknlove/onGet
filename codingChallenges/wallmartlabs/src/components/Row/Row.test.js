import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Row from './index';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('row component', () => {
  it('matches snapshot', function() {
    let wrapper = shallow(<Row />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('renders without crashing', function() {
    let wrapper = mount(<Row />);
    console.log(wrapper.debug());
  });
});
