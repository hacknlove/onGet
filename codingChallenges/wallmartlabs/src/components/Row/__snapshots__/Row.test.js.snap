// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row component matches snapshot 1`] = `
<div
  className="row"
>
  <div
    className="row-title"
  />
  <div
    className="row-figure"
    style={
      Object {
        "color": "black",
      }
    }
  >
    $
  </div>
</div>
`;
